draft-rescorla-tls-esni.md

```diff
@@ -402,20 +396,29 @@ serves the connection directly (if in Shared Mode), in which case
 it executes the steps in the following section, or forwards
 the TLS connection to the hidden server (if in Split Mode).
 
+## Shared Mode Hidden Server Behavior
 
-## Hidden Server Behavior
+A server operating in Shared Mode uses PaddedServerNameList.sni as
+if it were the "server_name" extension to finish the handshake. It
+SHOULD pad the Certificate message with the padding(21) extension {{!RFC7685}}
+with at least as much padding as that which was used to pad the SNI.
```

Comment:
Shouldn't we pad the Certificate message so that it would become as large as the largest Certificate message that would be offered from the server's IP address?

The same goes for the provision in the Split Mode.

Reply:
I agree with what @kazuho says here. It's not straightforward.

it's worth noting that this is one case in which random padding might be of some value, because the attacker doesn't know which connections correspond to the same SNI, and therefore can't do simple averageing.

Reply:
@kazuho I like it -- that's more precise. I'll update accordingly.